include/triton/core/tritonserver.h

```diff
@@ -847,6 +857,24 @@ TRITONSERVER_DECLSPEC TRITONSERVER_Error* TRITONSERVER_InferenceTraceParentId(
 TRITONSERVER_DECLSPEC TRITONSERVER_Error* TRITONSERVER_InferenceTraceModelName(
     TRITONSERVER_InferenceTrace* trace, const char** model_name);
 
+/// Set the name associated with a trace.
+///
+/// \param trace The trace.
+/// \param trace_name The name associated with the trace.
+/// \return a TRITONSERVER_Error indicating success or failure.
+TRITONSERVER_DECLSPEC TRITONSERVER_Error* TRITONSERVER_InferenceTraceSetName(
+    TRITONSERVER_InferenceTrace* trace, const char* trace_name);
+
+/// Get the name associated with a trace. The caller does
+/// not own the returned string and must not modify or delete it. The
+/// lifetime of the returned string extends only as long as 'trace'.
+///
+/// \param trace The trace.
+/// \param trace_name Returns the name associated with the trace.
+/// \return a TRITONSERVER_Error indicating success or failure.
+TRITONSERVER_DECLSPEC TRITONSERVER_Error* TRITONSERVER_InferenceTraceName(
+    TRITONSERVER_InferenceTrace* trace, const char** trace_name);
```

Comment:
Looking at the usage, this is actually the name of the custom activity? I think you should let `TRITONSERVER_InferenceTraceReportActivity` accept the name as another parameter so the backend doesn't have to call the setter before calling the report function (implicitly set by Triton). Also make it clear that if the tracer receive custom activity, should call the getter function(s) for additional information regarding the activity

Reply:
> Looking at the usage, this is actually the name of the custom activity?

That's correct.

> I think you should let TRITONSERVER_InferenceTraceReportActivity accept the name as another parameter so the backend doesn't have to call the setter before calling the report function (implicitly set by Triton).

I think the way it is currently implemented would allow us to add more fields to the trace without having to modify the signature of `TRITONSERVER_InferenceTraceReportActivity`. Let me know if this makes sense to you.

> Also make it clear that if the tracer receive custom activity, should call the getter function(s) for additional information regarding the activity

👍

Reply:
> I think you should let TRITONSERVER_InferenceTraceReportActivity accept the name as another parameter so the backend doesn't have to call the setter before calling the report function (implicitly set by Triton).


I actually see the problem now. How about adding a `TRITONSERVER_InferenceTrace* trace_attributes` that currently only has a name?

Reply:
I don't know if there is a need to add another struct for this, when capturing a timestamp, what else do you need to know beside the timestamp itself and the activity it indicates? Also note that those should be properties that are activity specific (i.e. may be different on different capture of the timestamp), if they are properties of the entire trace, they should not be tied to this function (i.e. request ID / parent ID is always the same for the entire trace).

If you perceive more capture specific attribute may be added into the future, I think you should make it generic (string key value pair to be attached to the capture) so all parties are not tied to the specific getter/setter exposed in Triton API. For example, Triton user is likely going to define their own attributes for tracking backend model execution, which is going to be not useful if they have to go through the API change so that the tracer can extract the information. On the other hand, the tracer implemented can simply dump all attached key-value if any, without worrying to modify the implementation whenever a new attribute is added.